### PR TITLE
zotaut keywords 655

### PR DESCRIPTION
**Pull request by mtrojan-ub**

### CODE REVIEW

File: cpp/lib/src/FileUtil.cc [hunk 1]
```diff
@@ -92,6 +92,19 @@ ReadLines::ReadLines(const std::string &path, const TrimMode trim_mode): trim_mo
 }
 
 
+std::vector<std::string> ReadLines::ReadOrDie(const std::string &path, const ReadLines::TrimMode trim_mode) {
+    std::vector<std::string> lines;
```
Comment (ruschein) [inline cpp/lib/src/FileUtil.cc:96]: This should be declared as close to its first use as possible.

File: cpp/lib/src/ZoteroHarvesterConversion.cc [hunk 2]
```diff
@@ -391,6 +391,12 @@ void StripBlacklistedTokensFromAuthorName(std::string * const first_name, std::s
 }
 
 
+bool Is655Keyword(const std::string &keyword) {
+    static const auto keyword_list(FileUtil::ReadLines::ReadOrDie(UBTools::GetTuelibPath() + "zotero-enhancement-maps/keywords_as_655.txt"));
```
Comment (ruschein) [inline cpp/lib/src/ZoteroHarvesterConversion.cc:395]: That seems like a strange name "keywords_as_655.txt".

File: cpp/lib/src/ZoteroHarvesterConversion.cc [hunk 3]
```diff
@@ -1088,8 +1094,13 @@ void GenerateMarcRecordFromMetadataRecord(const MetadataRecord &metadata_record,
         marc_record->insertField("773", _773_subfields);
 
     // Keywords
-    for (const auto &keyword : metadata_record.keywords_)
-        marc_record->insertField(MARC::GetIndexField(TextUtil::CollapseAndTrimWhitespace(keyword)));
+    for (const auto &keyword : metadata_record.keywords_) {
+        const std::string normalized_keyword(TextUtil::CollapseAndTrimWhitespace(keyword));
+        if (Is655Keyword(normalized_keyword))
+            marc_record->insertField("655", { { 'a', normalized_keyword } });
```
Comment (ruschein) [inline cpp/lib/src/ZoteroHarvesterConversion.cc:1100]: There is a variant of insertField that is simpler.

**Review by ruschein — CHANGES_REQUESTED** [review]

File: cpp/lib/src/ZoteroHarvesterConversion.cc (hunk 2 above)
Comment (mtrojan-ub) [reply cpp/lib/src/ZoteroHarvesterConversion.cc:395]: What would be a better name?

Comment (ruschein) [reply cpp/lib/src/ZoteroHarvesterConversion.cc:395]: marc655_keywords.txt?

**Pull request merged**